duality-solutions/Dynamic
# BDAP v2.4.16 All Updates

-  Add ed25519 key unit tests.
-  Fix spork address for regtest network
-  Modify `makekeypair` RPC command, output compressed keys
-  Fix existing unit tests in key_tests.cpp
-  Add bdap_link_tests unit tests
-  Add dht_data_tests unit tests
-  Fix link status when to and from your wallet
-  Use `GetKeyFromPool` instead of `MakeNewKey` in BDAP RPC functions
-  Remove DHT calls to session pointer within session methods
-  Pass by constant reference in event classes
-  Add root OID registered with ANSI and OID generation method
-  Use bytes for DHT instead of strings for put and get operations. Increases max bytes per chunk from 450 to 900
-  Persist DHT put buffer so data doesn't get overwritten
-  Make sure wallet is not null before processing links in manager
-  Update VGP library sub-module with verify ciphertext checks
-  Fix link status when to and from your wallet
-  Return an error in `putbdapdata` when record is locked.
-  Add bdap_vgp_message_tests
-  Only Dynode respond to get and put storage requests
-  Fix depreciated warning when using bdecode
-  Add wallet addresses to CLink class
-  Handle invalid DHT header data better
-  Fix invalid sequence number when putting deny link list
